File: core/src/main/java/com/triangleleft/flashcards/ui/vocabular/IVocabularyListView.java
package com.triangleleft.flashcards.ui.vocabular;

import android.support.annotation.NonNull;

import com.triangleleft.flashcards.service.vocabular.VocabularyWord;
import com.triangleleft.flashcards.ui.common.view.IView;

import java.util.List;

public interface IVocabularyListView extends IView {

    void showWords(@NonNull List<VocabularyWord> words, int selectedPosition);

    void showProgress();

    void showRefreshError();

    void showLoadError();

    void showEmpty();
}

File: core/src/main/kotlin/com/triangleleft/flashcards/ui/vocabular/IVocabularyListView.kt
package com.triangleleft.flashcards.ui.vocabular

import com.triangleleft.flashcards.ui.ViewEvent
import com.triangleleft.flashcards.ui.common.view.IView
import io.reactivex.Observable

interface IVocabularyListView : IView<VocabularyListViewState, ViewEvent> {

    fun refreshes(): Observable<Unit>

    fun wordSelections(): Observable<VocabularyWordSelectEvent>

    // Can't we reuse refereshes here?
    fun loadListRetires(): Observable<Unit>

}
